eos/utils/kinematic.cc

```diff
@@ -34,6 +42,8 @@ namespace eos
         std::map<std::string, unsigned> variables_map;
 
         std::vector<std::string> variables_names;
+
+        std::vector<KinematicVariable> variables;
```

Comment:
Do we really need this extra variable? What about `variables_data`?

Reply:
We do have wrappers for forward iterators of type ```T_```, and indirect iterators that convert ```T_ *``` and ```std::shared_ptr<T_>``` iterators to iterators for ```T_```. We do not have iterators that
accept a lambda to turn 'nothing' into ```KinematicVariable```. This is also not costing much, since
it simply means one ```std::shared_ptr<>``` more per kinematic variable. If you have a better idea, I'm all ears!

Reply:
Alright. We can still improve this letter if there is a better idea. I don't have one now